sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -868,6 +868,29 @@ class SubquerySuite extends QueryTest with SharedSQLContext {
       Row(3, 3.0, 2, 3.0) :: Row(3, 3.0, 2, 3.0) :: Nil)
   }
 
+  test("SPARK-20690: Do not add missing attributes through subqueries") {
+    withTempView("onerow") {
+      Seq(1).toDF("c1").createOrReplaceTempView("onerow")
+
+      val e = intercept[AnalysisException] {
+        sql(
+          """
+            | select 1
+            | from   (select 1 from onerow t1 LIMIT 1)
```

Comment:
I'm surprised we support this syntax, I think the FROM clause must have an alias.

I checked with postgres, it will throw exception `subquery in FROM must have an alias`, can you check with other databases? Thanks!